repos/system_upgrade/common/actors/scanlvmconfig/libraries/scanner.py

```diff
@@ -0,0 +1,56 @@
+import os
+import re
+
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api
+from leapp.models import DistributionSignedRPM, LVMConfig, LVMConfigDevicesSection
+
+LVM_CONFIG_PATH = '/etc/lvm/lvm.conf'
+
+
+def _lvm_config_devices_parser(lvm_config_lines):
+    in_section = False
+    config = {}
+    for line in lvm_config_lines:
+        line = line.split("#", 1)[0].strip()
+        if "devices {" in line:
+            in_section = True
+            continue
+        if in_section and "}" in line:
+            in_section = False
+        if in_section:
+            if not line:
+                continue
+            value = line.split("=", 1)
+            config[value[0].strip()] = value[1].strip().strip('"')
+    return config
+
+
+def _read_config_lines(path):
+    with open(path) as lvm_conf_file:
+        return lvm_conf_file.readlines()
+
+
+def scan():
+    if not has_package(DistributionSignedRPM, 'lvm2'):
+        return
+
+    if not os.path.isfile(LVM_CONFIG_PATH):
+        api.current_logger().debug('The "{}" is not present on the system.'.format(LVM_CONFIG_PATH))
+        return
+
+    lvm_config_lines = _read_config_lines(LVM_CONFIG_PATH)
+    devices_section = _lvm_config_devices_parser(lvm_config_lines)
+
+    lvm_config_devices = LVMConfigDevicesSection(use_devicesfile=int(get_source_major_version()) > 8)
+    if 'filter' in devices_section:
+        lvm_config_devices.filter = re.findall(r'"([^"]*)"', devices_section['filter'])
+
+    if 'devicesfile' in devices_section:
+        lvm_config_devices.devicesfile = devices_section['devicesfile']
+
+    if 'use_devicesfile' in devices_section and devices_section['use_devicesfile'] in ['0', '1']:
```

Comment:
I don't think this condition is necessary.
```suggestion
    if 'use_devicesfile' in devices_section:
```